src/backend/sgx/mod.rs

```diff
@@ -109,91 +126,64 @@ impl crate::backend::Backend for Backend {
         code: Component,
         _sock: Option<&Path>,
     ) -> Result<Arc<dyn Keep>> {
-        // Calculate the memory layout for the enclave.
-        let layout = crate::backend::sgx::shim::Layout::calculate(
-            shim.region().into(),
-            code.region().into(),
-        );
+        // Find the offset for loading the code.
+        let slot = Span::from(shim.find_header(PT_ENARX_CODE).unwrap().vm_range());
```

Comment:
Maybe add an anyhow context to the error and return with `?` instead of `unwrap()`. This makes debugging a little bit easier.

Same for the rest.

Reply:
I can do that now. However, my next change is to pull out the common behavior between the backends and make the `Backend` API more intelligent. This was going to include better error handling processes. Do you want to wait for that? Or do you want me to improve error handling now?